pages/docs/data-modeling/model-joins.mdx

```diff
@@ -0,0 +1,91 @@
+import { Callout } from "nextra-theme-docs";
+
+# Model Joins
+
+In many cases, it can be useful to reuse data from one data model in others. This can be done in Glean by configuring joins across models. In essence, this is defining a [primary-foreign key relationship](https://docs.oracle.com/cd/E12100_01/books/admintool/admintool_DataModeling4.html#:~:text=A%20primary%20key%2Dforeign%20key,key%20columns%20in%20another%20table.) across two Glean models.
+
+After configuring a join, you'll be able to use any attributes from the joined model in the base model. Note that joins only include the model **attributes**, not the model **metrics**.
+
+<Callout type="info">
+  Currently, Glean joins are always LEFT OUTER JOINS. Any unmatched rows from the joined model will simply have blank values for the attributes.
+</Callout>
+
+## Setting up
+
+There are two steps to setting up a join between two models:
+
+1. Configuring a primary key on the joined model (e.g. Customers)
+2. Configuring a foreign key on the base model (e.g. Sales)
+
+### 1. Configuring a primary key
+
+You first need to configure a primary key on the model you wish to join. Primary keys must meet the following conditions:
+
+- They must be **unique** in the underlying data table - we validate this when you set it.
+- They must have a **string** or **numeric** type.
+
+You can configure a primary key using the following steps:
+
+1. Navigate to the model builder for the relevant model.
+
+2. Ensure you have defined an attribute for the primary key to be assigned to.
+
+3. Open the attribute editor and click `Set as primary key`. Uniqueness will also be validated at this step.
+
+<img
+  src="/model_joins/pk.png"
+  alt="Setting a primary key"
+  style={{ padding: "20px" }}
+/>
```

Comment:
this image doesn't actually highlight where the checkbox is. Draw a box over it or something?